### PR TITLE
Tabs icon toggle

### LINKED ISSUE #25552: Update TabList/Tab to toggle outline vs fill icon

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Changes
- Added support in tab styles for toggling between regular and filled icons
- Updated storybook to demonstrate using bundled icons

## Issues

Fixes #25552

## Screenshots

image

### CODE REVIEW

File: packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts [hunk 1]
```diff
@@ -16,6 +16,13 @@ const reservedSpaceClassNames = {
   content: 'fui-Tab__content--reserved-space',
 };
 
+// These should match the constants defined in @fluentui/react-icons
+// This package avoids taking a dependency on the icons package for only the constants.
+const iconClassNames = {
```
Comment (GeoffCoxMSFT) [inline packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts:21]: I think react-tabs should avoid a dependency on react-icons at least for now so I'm willing to manage declaring the constant here.  Filed this issue to track fixing it: https://github.com/microsoft/fluentui/issues/25598

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.941 kB`<br />`17.678 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.948 kB`<br />`52.993 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.446 kB`<br />`11.033 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d5d03853c4aabcf7cedd715caddd987ba44c1a60'>d5d03853c4aabcf7cedd715caddd987ba44c1a60</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts [hunk 2]
```diff
@@ -378,6 +391,14 @@ const useIconStyles = makeStyles({
     height: '24px',
     width: '24px',
   },
+  selected: {
+    [`& .${iconClassNames.filled}`]: {
+      display: 'inline',
+    },
+    [`& .${iconClassNames.regular}`]: {
```
Comment (smhigley) [inline packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts:398]: This makes me a bit worried, but your reasoning does make sense and I suppose it'd be easy enough to take a dependency on react-icons in the future if it becomes a problem.

**Review by smhigley — APPROVED** [review]

File: packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts (hunk 2 above)
Comment (GeoffCoxMSFT) [reply packages/react-components/react-tabs/src/components/Tab/useTabStyles.ts:398]: Yes - always easier to add the dependency than take it away :)

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/Avatar-mount_base.html">1271</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/Avatar-mount.html">1261</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/Button-mount_base.html">930</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/Button-mount.html">936</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/FluentProvider-mount_base.html">1493</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/FluentProvider-mount.html">1478</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/FluentProviderWithTheme-mount_base.html">580</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/FluentProviderWithTheme-mount.html">587</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">542</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">544</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">576</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">570</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/MakeStyles-mount_base.html">1954</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/MakeStyles-mount.html">1961</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/SpinButton-mount_base.html">2339</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25597/perf-test-react-components/SpinButton-mount.html">2338</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Pull request merged**